datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActions.java

```diff
@@ -89,10 +89,16 @@ protected void prepareExecution(DatalakeTriggerRestoreEvent payload, Map<Object,
 
             @Override
             protected void doExecute(SdxContext context, DatalakeTriggerRestoreEvent payload, Map<Object, Object> variables) {
-                LOGGER.info("Triggering datalake restore for {}", payload.getResourceId());
-
+                Long sdxId = payload.getResourceId();
+                // When SDX is created as part of re-size flow chain, SDX in payload will not have the correct ID.
+                if (!Strings.isNullOrEmpty(payload.getSdxName())) {
+                    SdxCluster sdxCluster = sdxService.getByNameInAccount(context.getUserId(), payload.getSdxName());
+                    context.setSdxId(sdxCluster.getId());
+                    sdxId = sdxCluster.getId();
+                }
+                LOGGER.info("Triggering datalake restore for {}", sdxId);
```

Comment:
How about something in the below lines? This way we are making sure that logic is hot only for resize not for other cases.              
  Long sdxId = payload.getResourceId();
                FlowLog flowLog = flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId()).get();
                FlowChainLog flowChainLog = flowchainLogService.findFirstByFlowChainIdOrderByCreatedDesc(flowLog.getFlowChainId()).get();
                if(flowChainLog.getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getName())) {
                    SdxCluster sdxCluster = sdxService.getByNameInAccount(context.getUserId(), payload.getSdxName());
                    payload.getResourceId(sdxCluster.getId());
                    sdxId = sdxCluster.getId();
                }

Reply:
This will avoid changes to the rest of the file.

Reply:
the payload doesn't have a way to override the resourceId, add that seems like a much bigger change than referencing the new sdxId throughout the method

Reply:
I see what you mean. That will affect all the events, right.

Reply:
Just this event, the next event in the chain take the sdxId from the datalakedr response